presto-main/src/main/java/io/prestosql/server/security/KerberosAuthenticator.java

```diff
@@ -64,7 +64,9 @@ public KerberosAuthenticator(KerberosConfig config)
         System.setProperty("java.security.krb5.conf", config.getKerberosConfig().getAbsolutePath());
 
         try {
-            String hostname = InetAddress.getLocalHost().getCanonicalHostName().toLowerCase(Locale.US);
+            String hostname = Optional.ofNullable(config.getHostName())
+                    .orElseGet(() -> getLocalHost().getCanonicalHostName()).toLowerCase(Locale.US);
```

Comment:
nit: this would be more readable

```suggestion
                    .orElseGet(() -> getLocalHost().getCanonicalHostName())
                    .toLowerCase(Locale.US);
```

Reply:
Agreed with the nit. Fixing.

Waiting on @electrum's opinion with regards to the configuration parameter name, but I see your point with regards to the ambiguity.